azkaban-hadoop-security-plugin/src/main/java/azkaban/security/AbstractHadoopSecurityManager.java

```diff
@@ -392,6 +394,20 @@ public void prefetchToken(final File tokenFile, final Props props, final Logger
    */
   protected void doPrefetch(final File tokenFile, final Props props, final Logger logger,
       final String userToProxy) throws HadoopSecurityManagerException {
+    final boolean addGroupAndUserForEffectiveUser =
+        props.getBoolean(AZKABAN_ADD_GROUP_AND_USER_FOR_EFFECTIVE_USER, false);
+    if (addGroupAndUserForEffectiveUser) {
+      // If linux group and user needs to be added for effectiveUser before job process starts
+      // then set it up before effective user is used to set permission for any file.
+      logger.info("Adding group and user for effective user: " + userToProxy);
+      try {
+        ExecuteAsUserUtils.addGroupAndUserForEffectiveUser(executeAsUser, userToProxy);
```

Comment:
Isn't the effective user same as user to proxy? Why trying to do add the same user from 2 different places?

Reply:
It is same as user to proxy. Moving this logic to JobRunner so that we won't need it at two places to cover various jobtypes.